lib/gen-name-dir-key.js

```diff
@@ -0,0 +1,6 @@
+'use strict';
+var cacheKey = require('./cache-key');
+
+module.exports = function genNameDirKey(name, dir) {
```

Comment:
I don't really grok why we need this file, but the implementation could be simplified to:

```suggestion
module.exports = cacheKey;
```

Reply:
I originally had genNameDirKey and genModulePathKey (or whatever it was) in the index.js file. I abstracted away from 'cacheKey' because that name was so general and non-obvious. I'm a bit unclear on why we need to have cache keys instead of using the original strings for name+dir, path and realFilePath. Any background either of you can give me on why?

Reply:
I ended up just going back to using 'cacheKey' in both places, eliminating the separate functions, and adding a short comment to cache-key.js. I still have the question about why we need to cache the key instead of the raw string, but am otherwise good to go.